velox/exec/Window.cpp

```diff
@@ -505,9 +511,14 @@ RowVectorPtr Window::getOutput() {
     result->childAt(i) = output;
   }
 
+  // Partial partitions be processed for StreamingWindowBuid.
+  // So we need to split the result based on the number of processedRows.
+  auto preProcessedRows = numProcessedRows_;
```

Comment:
@aditi-pandit Still need this minor change to split the result based on the actual processed row numbers. Do you have any suggestion? Thanks.

Reply:
@JkSelf : My understanding of what you are saying is 
i) First addInput() -> 24 rows partitionStartRows_ = [0, 8, 16, 24]. At this point numRows = 24, numProcessedRows_ = 0
ii) First getOutput() -> Should return above 24 rows numRows = 24, numProcessedRows_ = 24
iii) Second addInput() -> Last 16 rows. numRows =  40, numProcessedRows_ = 24
iv) Second getOutput -> Should return above 16 rows numRows = 40, numProcessedRows_ = 40

You are saying in ii) 40 is returned and not 24 ? The first getOutput() call should return numRowsLeft = numRows_ - numProcessedRows_ which is 24 at that time. 

Is there some misunderstanding ?

Reply:
Yes. I updated the code. Can you help to review again? Thanks.

Reply:
> @JkSelf : My understanding of what you are saying is i) First addInput() -> 24 rows partitionStartRows_ = [0, 8, 16, 24]. At this point numRows = 24, numProcessedRows_ = 0 ii) First getOutput() -> Should return above 24 rows numRows = 24, numProcessedRows_ = 24 iii) Second addInput() -> Last 16 rows. numRows = 40, numProcessedRows_ = 24 iv) Second getOutput -> Should return above 16 rows numRows = 40, numProcessedRows_ = 40
> 
> You are saying in ii) 40 is returned and not 24 ? The first getOutput() call should return numRowsLeft = numRows_ - numProcessedRows_ which is 24 at that time.
> 
> Is there some misunderstanding ?

@aditi-pandit This [change ](https://github.com/facebookincubator/velox/pull/5437#discussion_r1322440399) ensures that the numRows is 24 after the first addInput() operation, whereas it would be 40 without this modification.

Reply:
@aditi-pandit Do you have any further comment for this change? Thanks.

Reply:
I think I now get what is happening.

So basically, some input is received, and the last partition seen so far is not finalized. So although numRows = 40 only 24 can be output at the first getOutput.

A bit cleaner option is to track (and return) in callApplyLoop the number of rows for the current output block. Please try https://github.com/facebookincubator/velox/pull/6584.

If that works for you, then I'll get that PR submit.